Cargo.toml

```diff
@@ -36,7 +35,7 @@ crossbeam = "0.8"
 criterion = "0.4.0"
 
 [profile.release]
-lto = "fat"
+lto = "thin"
```

Comment:
I would like to see some benchmarks before we change it.
The blog post says that Thin LTO can *sometimes* be faster than Fat LTO.

> This LTO variant supports much better parallelism than fat LTO. It can achieve similar performance improvements as fat LTO (sometimes even better!), while taking much less total time by taking advantage of more CPUs.

https://blog.rust-lang.org/inside-rust/2020/06/29/lto-improvements.html

Reply:
[link to discussion](https://www.reddit.com/r/rust/comments/sct44o/is_it_true_that_thinlto_can_make_more_performant/)

Reply:
> Note that rust's thin-local LTO will optimize worse than codegen-units = 1 though.
> rust's thin-local LTO doesn't do anything when codegen-units = 1.

https://old.reddit.com/r/rust/comments/lyck1u/compiling_for_maximum_performance/
> So set codegen-unit=1 with lto=true may squeeze extra optimizations.

We should have `codegen-unit=1` and keep Fat LTO. From these discussions, that's the most performant combo.